src/commands/basemaps-github/__test__/create-pr.test.ts

```diff
@@ -0,0 +1,99 @@
+import { randomUUID } from 'node:crypto';
+import { afterEach, describe, it } from 'node:test';
+
+import { ConfigTileSetRaster, TileSetType } from '@basemaps/config/build/config/tile.set.js';
+import { EpsgCode, ImageFormat } from '@basemaps/geo';
+import { fsa } from '@chunkd/fs';
+import { StacCollection, StacVersion } from 'stac-ts';
+
+import { anyAsciiPrintableString, randomEnumValue, randomSetEntry } from '../../../utils/__test__/randomizers.js';
+import { anySlug } from '../../../utils/__test__/slugify.test.js';
+import { GithubApi } from '../../../utils/github.js';
+import {
+  basemapsCreatePullRequest,
+  linzBasemapsSourceCollectionRel,
+  validSourceBuckets,
+  validTargetBuckets,
+} from '../create-pr.js';
+import { randomArrayEntry } from './utils.js';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+const originalEnv = Object.assign({}, process.env);
+
+afterEach(() => {
+  process.env = originalEnv;
+});
+
+describe('basemapsCreatePullRequest.handler', () => {
+  it('should handle S3 target', async (t) => {
+    const targetUrl = `s3://${anyValidTargetBucket()}/${anyEpsgCode()}/${anySlug()}`;
+    t.mock.method(fsa, 'readJson', () => {
+      return Promise.resolve(anyStacCollection());
+    });
+    process.env['GITHUB_API_TOKEN'] = anyAsciiPrintableString();
+    t.mock.method(GithubApi.prototype, 'getContent', () => {
+      return Promise.resolve(Buffer.from(JSON.stringify(anyConfigTileSetRaster())));
+    });
+    t.mock.method(GithubApi.prototype, 'createBranch', () => {});
+    t.mock.method(GithubApi.prototype, 'createBlob', () => {});
+    t.mock.method(GithubApi.prototype, 'createCommit', () => {});
+    t.mock.method(GithubApi.prototype, 'updateBranch', () => {});
+    t.mock.method(GithubApi.prototype, 'createPullRequest', () => {});
+    const targetUrlsString = JSON.stringify([targetUrl]);
+
+    await basemapsCreatePullRequest.handler({
+      target: targetUrlsString,
+      repository: `${anyGitHubOwner()}/${anyGitHubRepositoryName()}`,
+    } as any);
+  });
+});
+
+function anyStacCollection(): StacCollection {
```

Comment:
I disagree. Random data for fields we *don't care about* should never break the tests (if they do, we have problems unrelated to the test). Using random data makes it abundantly clear which fields we *do* care about. In cases like this, where the architecture of the code means there is heaps of irrelevant data involved in the test, it's really useful to be able to easily ignore irrelevant parts of tests.

Reply:
While it is fun to play with random test data it can and will break tests in random ways that are impossible to reproduce unless you use a seeded random generator.

We have known data for these fields, Please use one of the many known examples for these tests.

Reply:
Our code is hard to test, because functional and side-effect-ful code are intermingled. Using random data makes the tests easier to understand, because it makes it easier to link the test setup to the assertion. If we use specific values for things we don't care about it becomes very difficult to work out which parts of the setup are relevant for the test - the user would basically have to modify one part of the input, run the test, and check whether it succeeds or fails. If it succeeds, the input is not relevant for the test, and if it fails the input *is* relevant for the test. They would then have to repeat for each input.

Again, if any of these tests fail with random data we probably have bigger issues, and should address those.

I do agree that we should use a seeded random generator though. Not sure how that's usually done in Node.js, but if it's anything like Python it should be pretty easy.